### DISCUSSION TITLE
Daily Script with Captcha

### DISCUSSION THREAD

**Discussion by hube-mic** [bimmerconnected/bimmer_connected#689] [Q&A] [answered]
Hi,

I running a daily python script to collect total kilometers of my car. I read all the posts/questions regarding new captcha but don't 100% understand.
My script is running on raspberry pi. How could I generate a captcha without the gui for the device? In windows it is working and then I can connect again (manually). Is there a solution for raspberry? Next step would be the automation again....

BR

**Comment by rikroe** [comment, marked as answer]
You can create the captcha on any system and copy it over. This is needed for the first time login only, afterwards you can just use the CLI as normal.

**Reply by hube-mic** [reply]
Thanks a lot for your fast reply to my message. You are right, I can use the same captcha. It is working with python on my PI. But CLI is not working on PI. In Windows it is working. Do you have any idea why not on Raspberry PI? I run the following command: bimmerconnected status --captcha-token $token $user $passwd "rest_of_world"
Errormessage:
![image](https://github.com/user-attachments/assets/3a9bab5a-c4f5-4997-b8eb-6955c3e76e8d)

**Reply by rikroe** [reply]
You can only use captcha token **once**. So if you have used for any login (doesn't matter if it succeeded or not) you have to get a new one.

**Reply by hube-mic** [reply]
OK. thanks working :-)
Any experience how long/often I can connect without captcha?

**Reply by rikroe** [reply]
Not yet, as this is very new.

With the latest update (0.17.2), it should break less often due to the way the credentials are stored.

**Reply by hube-mic** [reply]
Thanks a lot for your efforts and support. Until now it is running without additional captcha with version 0.17.1.

**Reply by rikroe** [reply]
Thanks for the feedback!
